### ISSUE TITLE
Develop an API between Proxy and Services

### ISSUE THREAD

**Issue by Christopher-Chianelli** [Christopher-Chianelli/PodIRCBot#3]
We need to develop an API between services and proxies:

- Proxy need to sends events in a way such that services can easily read them.
- Services need to be able to tell the Proxy to form arbitrary IRC commands.

Current proposal:

- Proxy and Services send messages as a JSON Object put on one line
- Proxy sends events from `irc-framework`, with a few additions
- Service sends events compatible with `irc-framework`, with an id so failure/success can be reported.

**Comment by AObuchow** [comment]
+1 for the current proposal.

Are [these the events](https://github.com/kiwiirc/irc-framework/blob/master/docs/events.md) compatible with `irc-framework` that you mention?

Also would the "few additions" be new events? Or would we add extra data to the event's that already exist.

**Comment by Christopher-Chianelli** [comment]
Yes; that is the exact irc-framework I am using. The few additions will be the following:

-shortNick: "cchianel" instead of "cchianel|wfh"
-botName: The name of the bot that got this event
-type: The kind of event
